code/game/objects/items/storage/belt.dm

```diff
@@ -310,9 +310,37 @@
 	max_storage_space = 20
 	can_hold = list("/obj/item/ammo_magazine/handful")
 
-	New()
-		select_gamemode_skin(type)
-		..()
+/obj/item/storage/belt/shotgun/New()
+	select_gamemode_skin(type)
+	..()
+
+/obj/item/storage/belt/shotgun/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/ammo_magazine/shotgun))
+		var/obj/item/ammo_magazine/shotgun/M = W
+		if(M.current_rounds)
+			if(contents.len < storage_slots)
+				to_chat(user, "<span class='notice'>You start refilling [src] with [M].</span>")
+				if(!do_after(user, 15, TRUE, 5, BUSY_ICON_GENERIC)) 
+					return
+
+				for(var/x = 1 to storage_slots)
+					if(M.current_rounds > 0)
+						var/obj/item/ammo_magazine/handful/new_handful = rnew(/obj/item/ammo_magazine/handful)
+						var/R = min(M.current_rounds, 5)
+						new_handful.generate_handful(M.default_ammo, M.caliber, 5, R, M.gun_type)
```

Comment:
You have to `M.update_icon(-R)`.
A cleaner way to handle this would be by modifying and using the existing proc `M.create_handful()`. If you make it return the handful if sucessful and FALSE if it fails to create one, or creates an empty one (that shouldn't happen), then you don't even need to modify further code.

Reply:
Create handful wouldn't work since it creates it creates it into a hand and this should work even if both of your hands are full.

Reply:
Not if you invoke it like I showed, without passing parameters. But modifying the proc as I described was part of the suggestion.